jobsdb/jobsdb.go

```diff
@@ -665,20 +691,36 @@ func (jd *HandleT) addNewDS(appendLast bool, insertBeforeDS dataSetT) dataSetT {
 		jd.assert(len(dList) > 0, fmt.Sprintf("len(dList): %d <= 0", len(dList)))
 		for idx, ds := range dList {
 			if ds.Index == insertBeforeDS.Index {
-				//We never insert before the first element
-				jd.assert(idx > 0, fmt.Sprintf("idx: %d <= 0", idx))
-				levels, levelVals := jd.mapDSToLevel(ds)
-				levelsPre, levelPreVals := jd.mapDSToLevel(dList[idx-1])
+				// We never insert before the first element
```

Comment:
@soumyadebm @SumanthPuram @chandumlg This change is regarding ds numbering. Please review this.